packages/crypto/src/crypto.ts

```diff
@@ -466,3 +568,251 @@ const bigIntToHex = (num: bigint, length: number): string => {
+    throw new Error(
+      "failed to convert DER-encoded signature: invalid tag for r"
+    );
+  }
+  index++; // Move past the INTEGER tag
+  const rLength = derSignatureBuf[index];
+  index++; // Move past the length byte
+  const r = derSignatureBuf.slice(index, index + rLength!);
+  index += rLength!; // Move to the start of s
+
+  // Parse 's' and check for integer tag (0x02)
+  if (derSignatureBuf[index] !== 0x02) {
+    throw new Error(
+      "failed to convert DER-encoded signature: invalid tag for s"
+    );
+  }
+  index++; // Move past the INTEGER tag
+  const sLength = derSignatureBuf[index];
+  index++; // Move past the length byte
+  const s = derSignatureBuf.slice(index, index + sLength!);
+
+  // Normalize 'r' and 's' to 32 bytes each
+  const rPadded = normalizePadding(r, 32);
+  const sPadded = normalizePadding(s, 32);
+
+  // Concatenate and return the raw signature
+  return new Uint8Array([...rPadded, ...sPadded]);
+};
+
+/**
+ * Loads an ECDSA public key from a raw format for signature verification.
+ *
+ * @param {Uint8Array} quorumPublic - The raw public key bytes.
+ * @returns {Promise<CryptoKey>} - The imported ECDSA public key.
+ */
+const loadQuorumKey = async (quorumPublic: Uint8Array): Promise<CryptoKey> => {
+  return await crypto.subtle.importKey(
+    "raw",
+    quorumPublic,
+    {
+      name: "ECDSA",
+      namedCurve: "P-256",
+    },
+    true,
+    ["verify"]
+  );
+};
+
+/**
+ * Decodes a private key based on the specified format.
+ *
+ * @param {string} privateKey - The private key to decode.
+ * @param {string} keyFormat - The format of the private key (e.g., "SOLANA", "HEXADECIMAL").
+ * @returns {Uint8Array} - The decoded private key.
+ */
+
+export const decodeKey = (privateKey: string, keyFormat: any): Uint8Array => {
+  switch (keyFormat) {
+    case "SOLANA":
+      const decodedKeyBytes = bs58.decode(privateKey);
+      if (decodedKeyBytes.length !== 64) {
+        throw new Error(
+          `invalid key length. Expected 64 bytes. Got ${decodedKeyBytes.length}.`
+        );
+      }
+      return decodedKeyBytes.subarray(0, 32);
+    case "HEXADECIMAL":
+      if (privateKey.startsWith("0x")) {
+        return uint8ArrayFromHexString(privateKey.slice(2));
+      }
+      return uint8ArrayFromHexString(privateKey);
+    default:
+      console.warn(
+        `invalid key format: ${keyFormat}. Defaulting to HEXADECIMAL.`
+      );
+      if (privateKey.startsWith("0x")) {
+        return uint8ArrayFromHexString(privateKey.slice(2));
+      }
+      return uint8ArrayFromHexString(privateKey);
+  }
+};
+
+/**
+ * Encrypts a private key bundle using HPKE and verifies the enclave signature.
+ *
+ * @param {string} privateKey - The private key in hexadecimal format to be encrypted.
+ * @param {string} keyFormat - The format of the private key (e.g., "SOLANA", "HEXADECIMAL").
+ * @param {any} importBundle - The bundle containing the enclave's public key and data signature for verification.
```

Comment:
this should be a string. This function should JSON.parse the bundle and error out if it's an invalid bundle after asserting that the right properties are defined on it. That will let you avoid the "any" in here, and it'll also let you be more confident about what's in the bundle (you can early throw if there are non-empty keys for data, signature, or quorum key)